gpt_engineer/steps.py

```diff
@@ -119,7 +122,9 @@ def execute_workspace(ai: AI, dbs: DBs):
         return messages
     print('Executing the code...')
     print()
-    subprocess.run(command, shell=True)
+
+    # Run the subprocess in dbs.workspace.path
+    subprocess.run('bash run.sh', shell=True, cwd=dbs.workspace.path)
```

Comment:
Hey @AntonOsika 

I think I am missing something here, are we forcing the creation of a `run.sh` file somewhere?

Reply:
might need an environment flag to choose which shell to use for this. something like 

```python
import subprocess
import psutil
import os
#https://github.com/giampaolo/psutil
p = psutil.Process(os.getppid())
env = p.environ()
print(env)
if 'COMSPEC' in env:
    subprocess.run(f'cmd run.bat', shell=True, cwd=dbs.workspace.path)
if 'SHELL' in env:
    subprocess.run(f'{env['SHELL']} run.sh', shell=True, cwd=dbs.workspace.path) #{env['SHELL']}will return the shell like this '/bin/bash'

```

Reply:
Can I change this in my own build? Super unfamiliar with github and the like, but I'm trying to run gp-engineer through vscode terminal and conda env.